src/NotificationEventMailing.php

```diff
@@ -148,22 +148,10 @@ public static function send(array $data)
                 }
             }
 
-           // Add custom header for mail grouping in reader
-            $mmail->AddCustomHeader(
-                str_replace(
-                    [
-                        '%uuid',
-                        '%itemtype',
-                        '%items_id'
-                    ],
-                    [
-                        Config::getUuid('notification'),
-                        $current->fields['itemtype'],
-                        $current->fields['items_id']
-                    ],
-                    "In-Reply-To: <GLPI-%uuid-%itemtype-%items_id>"
-                )
-            );
+            // Add custom header for mail grouping in reader
+            $thread_id = $current->getItemMessageId('new');
```

Comment:
In some cases, the `new` event cannot be used as reference for a thread. For instance, the `checkpluginsupdate` event of `Glpi\Marketplace\Controller` class should probably use its own event name in thread id.

A way to handle this could be to have a `CommonDBTM::getMessageReferenceEvent(string $event): ?string` method that could be overridden by some classes.

```php
// CommonDBTM
public function getMessageReferenceEvent(string $event): ?string {
    $source_event = null;
    switch ($event) {
        case 'new':
        case 'update':
        case 'delete':
            // CRUD actions should refers to `new` thread
            $source_event = 'new';
            break;
        default:
            // Other actions should have their own thread
            $source_event = $event;
            break;
    }
    return $source_event;
}
// CommonITILObject
public function getMessageReferenceEvent(string $event): ?string {
    // All actions should refers to `new` thread
    return $event;
}
```

Reply:
@orthagh Did you missed this comment ?